### PR TITLE
Bump the default LTS from 11 to 14

### PR DESCRIPTION
**Pull request by snoyberg**
I'm fairly certain that we would also need to modify the integration
tests to use a more recent GHC, but not sure on procedure there.

Note: Documentation fixes for https://docs.haskellstack.org/en/stable/ should target the "stable" branch, not master.

Please include the following checklist in your PR:

* [ ] Any changes that could be relevant to users have been recorded in the ChangeLog.md
* [ ] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

**Comment by borsboom** [comment]
Have all the implications of this, in particular for the release process, been thought through?  For example, we'll need to ensure that all platforms we wish to target for binary releases are still supported.  There will be a fair amount of adjustment needed to the release process to work with this update.  In the past we've tended to do that adjustment after we decide it's time for a release, but that's caused significant delays when problems are discovered, so it might be a better idea to run through the whole release process before merging this PR.

I see that `stack-lts-12.yaml` is still in place.  Is the intent to keep that there to continue supporting GHC 8.4 as well?

The integration tests should all be updated to a matching LTS.  Otherwise, _every_ integration test will end up pulling and building the snapshot from scratch, which will greatly increase the amount of time they take.  Either that or the integration test framework should be updated to better handle multiple snapshots.

**Comment by snoyberg** [comment]
> so it might be a better idea to run through the whole release process before merging this PR.

Seems reasonable to me.

> I see that stack-lts-12.yaml is still in place. Is the intent to keep that there to continue supporting GHC 8.4 as well?

That was my initial thought, but I'm more than happy to drop it instead.

**Comment by borsboom** [comment]
I'm not sure if this is still the case with the Azure Pipelines CI, but back when CI was done on Travis and others there were also some changes needed to the matrix so that the right GHC version would be pre-installed.  Maybe @psibi can weight in on whether that's necessary here?

**Comment by borsboom** [comment]
I did a sweep through all the platforms, and it looks GHC 8.6.5 doesn't have any ARMv7 bindists.  Are we OK with dropping ARMv7 bindists for the next major Stack release?  I do think ARMv7 users appreciate having the bindist, just because it takes _so_ long to build to build from source on that platform.  That said, given no recent GHC versions have ARMv7 bindists (GHC 8.4.4 doesn't either), I don't think it would be unreasonable to drop support.

**Comment by snoyberg** [comment]
I think under the circumstances it's reasonable to drop ARMv7 support.

File: snapshot.yaml [hunk 1]
```diff
@@ -1,35 +1,13 @@
...
-- persistent-sqlite-2.9.3@rev:0
-- rio-0.1.11.0@rev:0
-- ansi-wl-pprint-0.6.8.2@rev:1 # for ansi-terminal-0.9
-- hedgehog-0.6.1@rev:4 # for ansi-terminal-0.9
-- typed-process-0.2.6.0@rev:0 # for rio-0.1.10.0
-- rio-prettyprint-0.1.0.0@rev:0
-- hi-file-parser-0.1.0.0@rev:0
-- http-download-0.1.0.0@rev:0
+- hpack-0.33.0@rev:0
 - github: snoyberg/filelock
```
Comment (borsboom) [inline snapshot.yaml]: Is there any hope of someday being able to use the Hackage version of `filelock`?

Comment (snoyberg) [reply snapshot.yaml]: I've pinged @nh2 about this, hopefully the changes will be merged upstream.

**Comment by snoyberg** [comment]
@jeffhappily will you take a stab at resolving the issues with this PR that @borsboom pointed out? Specifically, we need to update the resolvers used by the integration tests to use GHC 8.6.5, or theoretically modify the integration test runner to not require reinstalling GHC each time.

**Comment by jeffhappily** [comment]
No prob, I'll work on it

File: snapshot.yaml (hunk 1 above)
Comment (nh2) [reply snapshot.yaml]: I have commented on https://github.com/takano-akio/filelock/pull/7 for some feedback from @takano-akio.

**Pull request merged**